orca-retrofit/src/test/java/com/netflix/spinnaker/orca/retrofit/exceptions/SpinnakerServerExceptionHandlerTest.java

```diff
@@ -153,12 +159,82 @@ private static Stream<Arguments> exceptionsForHandlesTest() {
+                    .build())
+            .addCallAdapterFactory(ErrorHandlingExecutorCallAdapterFactory.getInstance())
+            .addConverterFactory(JacksonConverterFactory.create())
+            .build();
+
+    // HTTP 503 is always retryable
+    retrofit2.Response response =
+        retrofit2.Response.error(HttpStatus.SERVICE_UNAVAILABLE.value(), getDummyResponseBody());
+    SpinnakerServerException retryable = new SpinnakerHttpException(response, retrofit2Service);
+
+    // not retryable because POST is not idempotent
+    okhttp3.Request notRetryableRequest =
+        new okhttp3.Request.Builder().url(URL).post(getDummyRequestBody()).build();
+
+    retrofit2.Response notRetryableResponse =
+        makeRetrofit2Response(notRetryableRequest, HttpStatus.BAD_REQUEST.value());
+    SpinnakerServerException notRetryable =
+        new SpinnakerHttpException(notRetryableResponse, retrofit2Service);
+
+    // not retryable despite idempotent method because response code not within 429, 502, 503, 504
+    okhttp3.Request notRetryableIdempotentRequest =
+        new okhttp3.Request.Builder().url(URL).get().build();
+
+    retrofit2.Response notRetryableIdempotentResponse =
+        makeRetrofit2Response(notRetryableIdempotentRequest, HttpStatus.UNAUTHORIZED.value());
+    SpinnakerServerException notRetryableIdempotent =
+        new SpinnakerHttpException(notRetryableIdempotentResponse, retrofit2Service);
+
+    // idempotent network errors are retryable
+    okhttp3.Request retryableRequest = new okhttp3.Request.Builder().url(URL).get().build();
+
+    Throwable cause = new Throwable("error");
+    SpinnakerServerException idempotentNetworkException =
+        new SpinnakerNetworkException(cause, retryableRequest);
+
+    // throttling is retryable if idempotent
+    okhttp3.Request throttledIdempotentRequest =
+        new okhttp3.Request.Builder().url(URL).get().build();
+
+    retrofit2.Response throttledIdempotentResponse =
+        makeRetrofit2Response(throttledIdempotentRequest, HttpStatus.TOO_MANY_REQUESTS.value());
+    SpinnakerServerException throttledIdempotent =
+        new SpinnakerHttpException(throttledIdempotentResponse, retrofit2Service);
+
+    // throttling is not retryable if not idempotent
+    okhttp3.Request throttledNonIdempotentRequest =
+        new okhttp3.Request.Builder().url(URL).post(getDummyRequestBody()).build();
+
+    retrofit2.Response throttledNonIdempotentResponse =
+        makeRetrofit2Response(throttledNonIdempotentRequest, HttpStatus.TOO_MANY_REQUESTS.value());
+    SpinnakerServerException throttledNonIdempotent =
+        new SpinnakerHttpException(throttledNonIdempotentResponse, retrofit2Service);
+
+    return Stream.of(
+        Arguments.of(retryable, true),
+        Arguments.of(notRetryable, false),
+        Arguments.of(notRetryableIdempotent, false),
+        Arguments.of(idempotentNetworkException, true),
+        Arguments.of(throttledIdempotent, true),
```

Comment:
This test fails before the code change.  As in, `throttledIdempotent` is not considered retryable before the code change.  Same with `idempotentNetworkException`.

I think we need the exception to happen as the result of an actual call to a retrofit2 class to see, instead of building the exception class manually.  I bet something the Retrofit2Service class from [here](https://github.com/spinnaker/kork/blob/e823562f7dd449b980708c1562b718c1e9da70e1/kork-retrofit/src/test/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerRetrofit2ErrorHandleTest.java#L69) and [here](https://github.com/spinnaker/kork/blob/e823562f7dd449b980708c1562b718c1e9da70e1/kork-retrofit/src/test/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerRetrofit2ErrorHandleTest.java#L187) would do it?

Reply:
yeah, I was trying to avoid the extra setup but looks like it can't be helped. Should I go ahead and do this a new branch with only the tests or keep it here?

Reply:
I'd like to keep it in this PR so all the comments/context stay together.  I'm happy to help with the git juggling if you'd like.

Reply:
I think those two exceptions are not being considered retryable because `findHttpMethodAnnotation` is looking for this [`RestMethod`](https://github.com/spinnaker/orca/blob/74ba9caa785f817e7f475d1994add5b9c5c5c4e4/orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/BaseRetrofitExceptionHandler.groovy#L74) annotation which is present in [retrofit](https://github.com/square/retrofit/blob/36eab5b67dbb4bf37c71a7d07ac571992b4916b2/retrofit/src/main/java/retrofit/http/GET.java#L29) but not [retrofit2](https://github.com/square/retrofit/blob/trunk/retrofit/src/main/java/retrofit2/http/GET.java), so it's getting null for the HTTP method. It seems that without any code changes all retrofit2 exceptions will be marked not retryable.

Reply:
Good point!  So then maybe this isn't a refactor, but a fix?  It's maybe a nit since I don't think orca actually uses retrofit2 anywhere yet.

I think still it would be better if the tests generated exceptions from "real" retrofit2 calls.  You agree?

Reply:
yeah, I agree on the test exceptions. Do you think it's worth updating `findHttpMethodAnnotation` to also work for retrofit2 exceptions on the off chance that `getHttpMethod` returns null?

Reply:
Nah, I think the way we're handing retrofit2 exceptions is sufficient.  From my reading of [the code](https://github.com/spinnaker/kork/blob/e823562f7dd449b980708c1562b718c1e9da70e1/kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerServerException.java), we set `httpMethod` every time from an okhttp3 Request object, and the [kotlin code for that](https://github.com/square/okhttp/blob/parent-4.9.3/okhttp/src/main/kotlin/okhttp3/Request.kt#L30) looks to me like it can't be null.